opensrp-app/src/main/java/org/smartregister/sync/helper/SyncServiceHelper.java

```diff
@@ -0,0 +1,433 @@
+package org.smartregister.sync.helper;
+
+import android.content.Context;
+import android.content.Intent;
+import android.util.Pair;
+
+import com.google.gson.Gson;
+
+import org.apache.commons.lang3.StringUtils;
+import org.joda.time.DateTime;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.smartregister.AllConstants;
+import org.smartregister.CoreLibrary;
+import org.smartregister.R;
+import org.smartregister.SyncConfiguration;
+import org.smartregister.domain.FetchStatus;
+import org.smartregister.domain.Response;
+import org.smartregister.domain.Task;
+import org.smartregister.domain.db.EventClient;
+import org.smartregister.receiver.SyncStatusBroadcastReceiver;
+import org.smartregister.repository.EventClientRepository;
+import org.smartregister.service.HTTPAgent;
+import org.smartregister.util.NetworkUtils;
+import org.smartregister.util.SyncUtils;
+import org.smartregister.view.activity.DrishtiApplication;
+
+import java.text.MessageFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import timber.log.Timber;
+
+public class SyncServiceHelper {
+    private static final String ADD_URL = "/rest/event/add";
+    public static final String SYNC_URL = "/rest/event/sync";
+
+    private Context context;
+    private HTTPAgent httpAgent;
+    private SyncUtils syncUtils;
+
+    protected static final int EVENT_PULL_LIMIT = 250;
+    protected static final int EVENT_PUSH_LIMIT = 50;
+    protected static SyncServiceHelper instance;
+
+    public SyncServiceHelper() {
```

Comment:
This should be private if you are using the `singleton` pattern to get the instance